primitives/state-machine/src/ext.rs

```diff
@@ -354,17 +363,26 @@ where
 	}
 
 	fn place_storage(&mut self, key: StorageKey, value: Option<StorageValue>) {
-		trace!(target: "state", "{:04x}: Put {}={:?}",
-			self.id,
-			HexDisplay::from(&key),
-			value.as_ref().map(HexDisplay::from)
-		);
 		let _guard = guard();
 		if is_child_storage_key(&key) {
 			warn!(target: "trie", "Refuse to directly set child storage key");
 			return;
 		}
 
+		trace!(
+			target: "state",
+			method = "Put",
+			ext_id = self.id,
+			key = %HexDisplay::from(&key),
+			result = %HexDisplay::from(&value.encode()),
+			result = %HexDisplay::from(
+				&value
+					.as_ref()
+					.map(|v| EncodeOpaqueValue(v.clone()))
+					.encode()
+			),
```

Comment:
```suggestion
			value = %value.as_ref().map(HexDisplay::from),
			value_encoded = %HexDisplay::from(
				&value
					.as_ref()
					.map(|v| EncodeOpaqueValue(v.clone()))
					.encode()
			),
```

Reply:
Not sure if it's ok to rename the keys here, cc @emostov

Reply:
@bkchr & @dvdplm 

I changed both to look like:

```rust
		trace!(
			target: "state",
			method = "Get",
			ext_id = self.id,
			key = %HexDisplay::from(&key),
			result = ?result.as_ref().map(HexDisplay::from),
			result_encoded = %HexDisplay::from(
				&result
					.as_ref()
					.map(|v| EncodeOpaqueValue(v.clone()))
					.encode()
			),
		);
		```

Reply:
However my preference would be leave out the non-encoded field since I think it would only be useful for visual inspection (and still it will probably just be an array of ints, really sure if thats helpful).

Reply:
See commit `1f3ad4e` for changes

Reply:
Once this is settled though looks ready to merge ?

Reply:
> However my preference would be leave out the non-encoded field since I think it would only be useful for visual inspection (and still it will probably just be an array of ints, really sure if thats helpful).

I have already used this quite often for debugging state mismatches :D And seeing `Some(hash)` or `None` is still better than parsing just the raw encoded data.

Reply:
@emostov why not call it here `value =` and `value_encoded`? As this is not the result in this context.

Reply:
@bkchr just trying to avoid extra logic on the processing side, but I will change it to your suggestion so it is more readable in context.


> I have already used this quite often for debugging state mismatches :D And seeing Some(hash) or None is still better than parsing just the raw encoded data.

I needed to change
```
result = ?result.as_ref().map(HexDisplay::from),
```
to use the `?` operator, which is tracing shorthand for `Debug`, should that be ok for your purposes? (Using `%` for `Display` had some issues and I didn't try for to long to sort it out)
[see here for tracing field shorthand docs](https://docs.rs/tracing/0.1.26/tracing/index.html#recording-fields)

Reply:
Yeah I know the tracing operators ;) 

It was actually a mistake by me to suggest `%`. Before we also used `Debug`, so everything fine now :)